cpp/src/Ice/Connection.cpp

```diff
@@ -32,3 +32,39 @@ Ice::Connection::flushBatchRequestsAsync(CompressBatch compress)
         [promise](bool) { promise->set_value(); });
     return promise->get_future();
 }
+
+future<void>
+Ice::Connection::close() noexcept
+{
+    auto sharedPromise = make_shared<promise<void>>();
+    close(
+        [sharedPromise](exception_ptr closeException)
+        {
+            try
+            {
+                rethrow_exception(closeException);
+            }
+            catch (const ConnectionClosedException&)
```

Comment:
We need to keep this list of "graceful close" exceptions in sync in all languages.

Reply:
Removed.